sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -59,6 +59,8 @@ object DateTimeUtils {
   final val toYearZero = to2001 + 7304850
   final val TimeZoneGMT = TimeZone.getTimeZone("GMT")
 
+  lazy val c = Calendar.getInstance(TimeZone.getTimeZone("GMT"))
```

Comment:
Yes if SimpleDateFormat isn't used in multiple threads and you're sure it can't be, then caching it in the instance is OK. 

This however isn't thread-safe. The Calendar would be shared across threads.